pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -204,6 +204,10 @@ func isRunningAndReady(pod *v1.Pod) bool {
 	return pod.Status.Phase == v1.PodRunning && podutil.IsPodReady(pod)
 }
 
+func isRunningAndAvailable(pod *v1.Pod, minReadySeconds int32) bool {
```

Comment:
Since this is statefulset helper, I'd just pass statefulset object and let the `minReadySeconds` extraction be handled inside this method.